### ISSUE TITLE
Nested Decision Task  in Fork-Join Task,conductor cannot run the task after the decision task

### ISSUE THREAD

**Issue by dechykong** [Netflix/conductor#644]
conductor cannot run the task（API6） after the decision task(API4)。

![Nested Decision Task  in Fork-Join Task](https://user-images.githubusercontent.com/17778174/42513654-0d6dd188-848a-11e8-82da-1856b29f034f.png)

I'm confuse that whether the conductor support the workflow definition like this.
here's the workflow json:
````
{
    "updateTime":1531196677024,
    "name":"test-joinAndDecision",
    "description":"test-joinAndDecision",
    "version":0,
    "tasks":[
        {
            "name":"task_0",
            "taskReferenceName":"fork_join",
            "type":"FORK_JOIN",
            "forkTasks":[
                [
                    {
                        "name":"task_1",
                        "taskReferenceName":"API1",
                        "inputParameters":{
                          
                        },
                        "type":"HTTP",
                        "startDelay":0
                    },
                    {
                        "name":"task_3",
                        "taskReferenceName":"decision",
                        "inputParameters":{
                            "case_value_param":"four"
                        },
                        "type":"DECISION",
                        "caseValueParam":"case_value_param",
                        "decisionCases":{
                            "four":[
                                {
                                    "name":"task_4",
                                    "taskReferenceName":"API4",
                                    "inputParameters":{                                      
                                    },
                                    "type":"HTTP",
                                    "startDelay":0
                                }
                            ],
                            "five":[
                                {
                                    "name":"task_5",
                                    "taskReferenceName":"API5",
                                    "inputParameters":{
                                        
                                    },
                                    "type":"HTTP",
                                    "startDelay":0
                                }
                            ]
                        },
                        "startDelay":0
                    },
                    {
                        "name":"task_8",
                        "taskReferenceName":"API6",
                        "type":"EVENT",
                        "startDelay":0,
                        "sink":"conductor"
                    }
                ],
                [
                    {
                        "name":"task_2",
                        "taskReferenceName":"API2",
                        "inputParameters":{
                            
                        },
                        "type":"HTTP",
                        "startDelay":0
                    }
                ]
            ],
            "startDelay":0
        },
        {
            "name":"task_7",
            "taskReferenceName":"join",
            "type":"JOIN",
            "startDelay":0,
            "joinOn":[
                "API6",
                "API2"
            ]
        }
    ],
    "schemaVersion":2
}
````

**Comment by dechykong** [comment]
I debug the code,this seems like a bug，API4 next task is always join task，API6 is lost. Because I don't kown   which task will be schedule after decision-task,I add a API6 to wait for it. Finally join the API6 and API2.The result confuse me,API6 is lost,will never be schedule.

**Comment by dileepkeely** [comment]
Can you please let me know how did you load the workflow DSL json to be instantiated by conductor? 
Also how did u register your tasks?

**Comment by neokyle1** [comment]
It sounds like this may be the same issue as https://github.com/Netflix/conductor/issues/881

**Comment by apanicker-nflx** [comment]
Fixed as part of #1449
